include/statespace/StateSpace.hpp

```diff
@@ -148,6 +148,25 @@ class StateSpace {
         return true;
     }
 
+    /// Given input state, convert it to dynamic vector representing state.
+    /// Length, values of state depend on the derived class
+    ///
+    /// \param state Input state
+    /// \param[out] state_vector State vector
+    /// \return True if conversion successful; false otherwise
+    virtual bool get_state_vector(
+        const State& state,
+        Eigen::VectorXd* state_vector) const = 0;
+
+    /// Converts state vector representation to state
+    ///
+    /// \param state State
+    /// \param state_vector State vector
+    /// \return True if conversion successful; false otherwise;
+    virtual bool from_state_vector(
```

Comment:
`vector_to_state`

Reply:
Actually I think this function should be a constructor for the `State` class. You can input a vector to construct the State